### PR TITLE
Issue #355, Old URL redirects.

### LINKED ISSUE #355: URL Redirect
- [ ] Port over the code from HN to Jupiter for Fedora3 uuid redirect

Items deposited in /migrated from HN will maintain their URL/Id structure. No work required. 
Any migrated object would have their handles updated already with the HN url. No redirect needed.

### PR DESCRIPTION
**Pull request by cwant**
Fixes #355 

## HydraNorth URL redirects
- [x] get '/files/:noid', to: 'redirect#hydra_north_item' # may have query string `?file=filename`
- [x] get '/downloads/:noid', to: 'redirect#hydra_north_item' # may have query string `?file=filename`
- [x] get '/files/:noid/:filename', to: 'redirect#hydra_north_file', constraints: { filename: /[^\/]+/ }
- [x] get '/collections/:noid', to: 'redirect#hydra_north_community_collection'

## Pre-HydraNorth URL redirects
- [x] get '/public/view/item/:uuid', to: 'redirect#fedora3_item'
- [x] get '/public/view/item/:uuid/:ds', to: 'redirect#fedora3_datastream'
- [x] get '/public/view/item/:uuid/:ds/:filename', to: 'redirect#fedora3_datastream', constraints: { filename: /[^\/]+/ }
- [x] get '/public/datastream/get/:uuid/:ds', to: 'redirect#fedora3_datastream'
- [x] get '/public/datastream/get/:uuid/:ds/:filename', to: 'redirect#fedora3_datastream',
                                                    constraints: { filename: /[^\/]+/ }
- [x] get '/public/view/collection/:uuid', to: 'redirect#fedora3_collection'
- [x] get '/public/view/community/:uuid', to: 'redirect#fedora3_community'
- [x] ~~get '/public/view/author/:username', to: 'redirect#no_longer_supported'~~
- [x] ~~get '/action/submit/init/thesis/:uuid', to: 'redirect#no_longer_supported'~~

### CODE REVIEW

File: app/controllers/redirect_controller.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                          file_set_id: file_set.id,
+                          file_name: CGI.escape(file_set.contained_filename)), status: :moved_permanently
+    else
+      # If file not found, redirect to item level
+      redirect_to item_url(item), status: :found
+    end
+  end
+
+  def no_longer_supported
+    # Ancient thesis deposit URL and authors no longer supported
```
Comment (murny) [inline app/controllers/redirect_controller.rb]: Is there a certain point where we just remove these? 

We been telling google/etc that authors has been `410 gone` for over 2 years now (and before that we were telling google it was `missing 404`)

Google is currently not indexing this (same with thesis deposit url). So I guess why we keep moving this technical debt into this new app?

Comment (pbinkley) [reply app/controllers/redirect_controller.rb]: Based on the two months of Apache logs we have, spiders are still hitting this pattern a little, mostly Bing (638 bot hits, 58 non-bot, almost all referred from faculty web pages). So yeah, this one isn't doing real people any good. Let me check the other legacy patterns in the logs and trim the dead wood.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -71,6 +71,10 @@ def render_404
     render file: 'public/404.html', status: :not_found, layout: false
   end
 
+  def render_410
+    render file: 'public/404.html', status: :gone, layout: false
```
[the remark this answers is not in the document]
Comment (cwant) [reply app/controllers/application_controller.rb]: We might want to handle this when we handle the rest of the error pages (Issue #5). The current way we are doing this is with unstyled static pages, but within the scope of #5 we might want to have it dynamically display the status code too.

[the remark this answers is not in the document]
Comment (pbinkley) [reply app/controllers/application_controller.rb]: That makes sense. I'll add a comment there. So this PR is ready to merge as far as I'm concerned.

**Comment by cwant** [comment]
I've removed the two obsolete routes, so this PR will need re-approval.

**Pull request merged**